test/simulator/timestamp/src/timestamp_manager.cpp

```diff
@@ -331,3 +332,51 @@ timestamp_manager::validate_commit_timestamp(session_simulator *session, uint64_
 
     return (0);
 }
+
+/*
+ * Validate the prepare timestamp. The constraints on the prepare timestamp are:
+ * - Cannot set the prepared timestamp if the transaction is already prepared.
+ * - Cannot set prepared timestamp more than once.
+ * - Commit timestamp should not have been set before the prepare timestamp.
+ * - Prepare timestamp must be greater than the latest active read timestamp.
+ * - Prepare timestamp canot be less than the stable timestamp unless rounding
+ *   the prepare timestamp is enabled.
+ */
+int
+timestamp_manager::validate_prepare_timestamp(session_simulator *session, uint64_t prepare_ts)
+{
+    /* Cannot set the prepared timestamp if the transaction is already prepared. */
+    if (session->is_txn_prepared())
+        WT_SIM_RET_MSG(
+          EINVAL, "Cannot set the prepared timestamp if the transaction is already prepared");
+
+    /* A prepared timestamp should have been set at this point. */
```

Comment:
Should "not" have been set?

Reply:
Updated in [3ef8706](https://github.com/wiredtiger/wiredtiger/pull/8390/commits/3ef8706f5b58071fe4febfa0fdc0caa7efc5969e)

Reply:
Nice catch